extensions/amp-access/0.1/test/test-access-expr.js

```diff
@@ -258,86 +263,110 @@ describe('evaluateAccessExpr', () => {
+    expect(evaluator.evaluate('NOT obj["a num"]', resp)).to.be.false;
+    expect(evaluator.evaluate('NOT obj["other"]', resp)).to.be.true;
+    expect(evaluator.evaluate('NOT obj["a other"]', resp)).to.be.true;
+
+    expect(evaluator.evaluate('obj2["bool"] = NULL', resp)).to.be.true;
   });
 
   it('should NOT evaluate nested expressions with wrong type', function() {
-    expect(evaluateAccessExpr('obj.bool = true', {obj: true})).to.be.false;
-    expect(evaluateAccessExpr('obj.num = 11', {obj: 11})).to.be.false;
-    expect(evaluateAccessExpr('obj.str = "A"', {obj: 'A'})).to.be.false;
-    expect(evaluateAccessExpr('obj.str = "A"', {})).to.be.false;
+    expect(evaluator.evaluate('obj.bool = true', {obj: true})).to.be.false;
+    expect(evaluator.evaluate('obj.num = 11', {obj: 11})).to.be.false;
+    expect(evaluator.evaluate('obj.str = "A"', {obj: 'A'})).to.be.false;
+    expect(evaluator.evaluate('obj.str = "A"', {})).to.be.false;
 
-    expect(evaluateAccessExpr('obj.other = NULL', {obj: 11})).to.be.true;
+    expect(evaluator.evaluate('obj.other = NULL', {obj: 11})).to.be.true;
 
     expect(() => {
-      evaluateAccessExpr('obj.NULL', {});
+      evaluator.evaluate('obj.NULL', {});
     }).to.throw();
     expect(() => {
-      evaluateAccessExpr('NULL.obj', {});
+      evaluator.evaluate('NULL.obj', {});
     }).to.throw();
     expect(() => {
-      evaluateAccessExpr('1.obj', {});
+      evaluator.evaluate('1.obj', {});
     }).to.throw();
     expect(() => {
-      evaluateAccessExpr('TRUE.obj', {});
+      evaluator.evaluate('TRUE.obj', {});
     }).to.throw();
   });
 
   it('should evaluate nested expressions securely', () => {
-    expect(evaluateAccessExpr('obj.__bool__ = NULL', {obj: {}})).to.be.true;
+    expect(evaluator.evaluate('obj.__bool__ = NULL', {obj: {}})).to.be.true;
   });
 
   it('should accept name grammar', () => {
-    expect(evaluateAccessExpr('num = 10', {num: 10})).to.be.true;
-    expect(evaluateAccessExpr('num1 = 10', {num1: 10})).to.be.true;
-    expect(evaluateAccessExpr('num_ = 10', {num_: 10})).to.be.true;
-    expect(evaluateAccessExpr('_num = 10', {_num: 10})).to.be.true;
+    expect(evaluator.evaluate('num = 10', {num: 10})).to.be.true;
+    expect(evaluator.evaluate('num1 = 10', {num1: 10})).to.be.true;
+    expect(evaluator.evaluate('num_ = 10', {num_: 10})).to.be.true;
+    expect(evaluator.evaluate('_num = 10', {_num: 10})).to.be.true;
 
     expect(() => {
-      evaluateAccessExpr('1num = 10', {'1num': 10});
+      evaluator.evaluate('1num = 10', {'1num': 10});
     }).to.throw();
     expect(() => {
-      evaluateAccessExpr('num-a = 10', {'num-a': 10});
+      evaluator.evaluate('num-a = 10', {'num-a': 10});
     }).to.throw();
     expect(() => {
-      evaluateAccessExpr('num-1 = 10', {'num-1': 10});
+      evaluator.evaluate('num-1 = 10', {'num-1': 10});
     }).to.throw();
   });
+
+  describe('caching', () => {
+    beforeEach(() => {
+      window.sandbox.spy(evaluator, 'eval_');
```

Comment:
Nit: Avoid inspecting private members if possible.

Reply:
I created this function _just_ for inspecting 🤔. Do you have any idea of how I could check how many times the internal `evaluateAccessExpr` is called?

Reply:
Make the property public.